zaza/openstack/charm_tests/ceph/osd/tests.py

```diff
@@ -47,3 +51,226 @@ def test_osd_security_checklist(self):
             expected_passes,
             expected_failures,
             expected_to_pass=True)
+
+
+class OsdService:
+    """Simple representation of ceph-osd systemd service."""
+
+    def __init__(self, id_: int):
+        """
+        Init service using its ID.
+
+        e.g.: id_=1 -> ceph-osd@1
+        """
+        self.id = id_
+        self.name = 'ceph-osd@{}'.format(id_)
+
+
+async def async_wait_for_service_status(unit_name, services, target_status,
+                                        model_name=None, timeout=2700):
+    """Wait for all services on the unit to be in the desired state.
+
+    Note: This function emulates the
+    `zaza.model.async_block_until_service_status` function, but it's using
+    `systemctl is-active` command instead of `pidof/pgrep` of the original
+    function.
+
+    :param unit_name: Name of unit to run action on
+    :type unit_name: str
+    :param services: List of services to check
+    :type services: []
```

Comment:
Yeah, I didn't want to make PR to zaza, given how close we were to 20.10 release. I'think this function could live in `zaza.model` module alongside the `async_block_until_service_status` and it could be called something like `async_block_until_systemd_service_status`.
While the original function is more portable, this one offers more reliability (if `systemd` is available)